File: C++/数学/Sieve_Eu.cpp
constexpr int MX = 1e8 + 1;
bool pd[MX];
vi primes;
auto init = [] {
    rep(i, 2, MX) {
        if (!pd[i]) primes.push_back(i);
        for (int& p : primes) {
            if (p * i > MX) break;
            pd[i * p] = true;
            if (i % p == 0) break;
        }
    }
    return 0;
}();

File: Python/数学/Sieve_Eu.py
MX = 6 * 10**6 + 1
pd = bytearray(MX)
primes = []

for i in range(2, MX):
    if not pd[i]:
        primes.append(i)
    for p in primes:
        if p * i >= MX:
            break
        pd[i * p] = True
        if i % p == 0:
            break
